modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/AmqpConnectionManager.java

```diff
@@ -31,7 +32,7 @@
     /**
      * List of {@link AmqpConnectionHandler} representing AMQP connections.
      */
-    private List<AmqpConnectionHandler> connectionHandlers;
+    private final List<AmqpConnectionHandler> connectionHandlers;
```

Comment:
When creating the connectionHandlers list shall we use synchronized list? That way we can get rid of the synchronized blocks in code.

```java
connectionHandlers = Collections.synchronizedList(new ArrayList<>());
```

Reply:
Can be solved by using the ConcurrentHashMap. Ignore the above